### PR TITLE
[TECH-DEBT] Add View Sync Timeout to Config

### LINKED ISSUE #1472: Add Timeout Config to View Sync Task
Consensus uses the timeout from the config for it's timer.  The view sync task should also have have a configurable timeout

### PR DESCRIPTION
**Pull request by bfish713**
Closes #1472 

### This PR: 

Adds a view sync timeout duration to the config and makes the default a less time than the view timeout.

### This PR does not: 

Shouldn't change anything else

### Key places to review: 

Are the defaults reasonable

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -84,45 +84,45 @@ toml = "0.8.11"
...
-        "relay",
-        "request-response",
-        "rendezvous",
-        "secp256k1",
-        "serde",
-        "tcp",
-        "uds",
-        "websocket",
-        "yamux",
+  "macros",
```
Comment (jparr721) [inline Cargo.toml:87]: What did you run to format this? Is it a vscode command?

File: crates/hotshot/src/traits/election/static_committee.rs [hunk 2]
```diff
@@ -80,9 +80,9 @@ where
     #[cfg(feature = "randomized-leader-election")]
     /// Index the vector of public keys with a random number generated using the current view number as a seed
     fn get_leader(&self, view_number: TYPES::Time) -> PUBKEY {
-        let mut rng: StdRng = rand::SeedableRng::seed_from_u64(*view_number as u64);
-        let randomized_view_number: u64 = rng.gen();
-        let index = (randomized_view_number % self.nodes_with_stake.len() as u64) as usize;
+        let mut rng: StdRng = rand::SeedableRng::seed_from_u64(*view_number);
+        let randomized_view_number: usize = rng.gen();
```
Comment (jparr721) [inline crates/hotshot/src/traits/election/static_committee.rs:84]: Nice.

**Review by jparr721 — APPROVED** [review]
This looks great. Though I think I'll need to re-approve after your types PR goes in.

File: Cargo.toml (hunk 1 above)
Comment (bfish713) [reply Cargo.toml:87]: Actually been trying out neovim and it was doing the toml formatting

**Comment by bfish713** [comment]
> This looks great. Though I think I'll need to re-approve after your types PR goes in.

Mind taking a look at the other PR?

File: Cargo.toml (hunk 1 above)
Comment (jparr721) [reply Cargo.toml:87]: Oooh I might bug you for your config later

**Review by jparr721 — APPROVED** [review]

**Pull request merged**